core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -721,6 +784,103 @@ private ExprValue exprWeek(ExprValue date, ExprValue mode) {
         CalendarLookup.getWeekNumber(mode.integerValue(), date.dateValue()));
   }
 
+  private ExprValue unixTimeStamp() {
+    return new ExprLongValue(Instant.now().getEpochSecond());
+  }
+
+  private ExprValue unixTimeStampOf(ExprValue value) {
+    var res = unixTimeStampOfImpl(value);
+    if (res == null) {
+      return ExprNullValue.of();
+    }
+    if (res < 0) {
+      // According to MySQL returns 0 if year < 1970, don't return negative values as java does.
+      return new ExprDoubleValue(0);
+    }
+    if (res >= MYSQL_MAX_TIMESTAMP) {
+      // Return 0 also for dates > '3001-01-19 03:14:07.999999' UTC (32536771199.999999 sec)
+      return new ExprDoubleValue(0);
+    }
+    return new ExprDoubleValue(res);
+  }
+
+  private Double unixTimeStampOfImpl(ExprValue value) {
+    // Also, according to MySQL documentation:
+    //    The date argument may be a DATE, DATETIME, or TIMESTAMP ...
+    switch ((ExprCoreType)value.type()) {
+      case DATE: return value.dateValue().toEpochSecond(LocalTime.MIN, ZoneOffset.UTC) + 0d;
+      case DATETIME: return value.datetimeValue().toEpochSecond(ZoneOffset.UTC)
+          + value.datetimeValue().getNano() / 1E9;
+      case TIMESTAMP: return value.timestampValue().getEpochSecond()
+          + value.timestampValue().getNano() / 1E9;
+      default:
+        //     ... or a number in YYMMDD, YYMMDDhhmmss, YYYYMMDD, or YYYYMMDDhhmmss format.
+        //     If the argument includes a time part, it may optionally include a fractional
+        //     seconds part.
+
+        var dateFormatShortYear = new DateTimeFormatterBuilder()
+            .appendValueReduced(YEAR, 2, 2, 1970)
+            .appendPattern("MMdd")
+            .toFormatter()
+            .withResolverStyle(ResolverStyle.STRICT);
+
+        var dateFormatLongYear = new DateTimeFormatterBuilder()
+            .appendValue(YEAR, 4)
+            .appendPattern("MMdd")
+            .toFormatter()
+            .withResolverStyle(ResolverStyle.STRICT);
+
+        var dateTimeFormatShortYear = new DateTimeFormatterBuilder()
+            .appendValueReduced(YEAR, 2, 2, 1970)
+            .appendPattern("MMddHHmmss")
+            .toFormatter()
+            .withResolverStyle(ResolverStyle.STRICT);
+
+        var dateTimeFormatLongYear = new DateTimeFormatterBuilder()
+            .appendValue(YEAR,4)
+            .appendPattern("MMddHHmmss")
+            .toFormatter()
+            .withResolverStyle(ResolverStyle.STRICT);
```

Comment:
Should we make all formatters static or move it to some utility class?

Reply:
We have class `OpenSearchDateFormatters`, but it is located in `opensearch` and not accessible in `core`. Do you want me to move it or use `DateTimeUtils` instead?

Reply:
Probably we can move it to `core` and use it to hold all common date formatters? Then `OpenSearchExprValueFactory` and others can choose which formatter is needed by themselves?